augur/cli/db.py

```diff
@@ -331,3 +332,188 @@ def check_pgpass_credentials(config):
+
+    repos_list = list(map(get_value_from_fetch_data, repos))
+
+    return repos_list
+
+def get_org_id(augur_app, org):
+    org_id_sql = s.sql.text("""
+            SELECT repo_group_id FROM repo_groups
+            WHERE rg_name = '{}'
+        """.format(org))
+
+    org_id = augur_app.database.execute(org_id_sql).fetchone()[0]
+    if type(org_id) != int:
+        logger.info("Failed to get organization id from org {}".format(org))
+    return org_id
+
+def get_all_repos(org):
+    totalCount = get_repos_total_count(org)
+    maxCount = 100
+    nodes = []
+    after = ""
+
+    while totalCount > maxCount or totalCount <= maxCount and totalCount > 0:
+        totalCount = totalCount - maxCount
+        query = """
+        {
+            organization(login: "%s") {
+                repositories(first: %d%s) {
+                    edges {
+                        node {
+                            id
+                            name
+                            nameWithOwner
+                            url
+                            createdAt
+                        }
+                        cursor
+                    }
+                }
+            }
+        }
+        """ % (org, maxCount, after)
+
+        data = get_data(query, org)
+        if 'data' not in data:
+            logger.info("Could not get a list of all repositories. 'data' is missing in: {}\n".format(data))
+        if 'organization' not in data['data']:
+            logger.info("Could not get a list of all repositories. 'organization' is missing in: {}\n".format(data))
+        if 'repositories' not in data['data']['organization']:
+            logger.info("Could not get a list of all repositories. 'repositories' is missing in: {}\n".format(data))
+        if 'edges' not in data['data']['organization']['repositories']:
+            logger.info("Could not get a list of all repositories. 'edges' is missing in: {}\n".format(data))
+
+        nodes = nodes + data['data']['organization']['repositories']['edges']
+
+        l = len(data['data']['organization']['repositories']['edges'])
+        last_el = data['data']['organization']['repositories']['edges'][l-1]
+        after = ", after: \"" + last_el['cursor'] + "\""
+
+    data['data']['organization']['repositories']['edges'] = nodes
+
+    return data['data']['organization']['repositories']['edges']
+
+def get_repos_total_count(org):
+    query = """
+    {
+        organization(login: "%s") {
+            repositories(first: 1) {
+                totalCount
+            }
+        }
+    }
+    """ % (org)
+
+    data = get_data(query, org)
+
+    return data['data']['organization']['repositories']['totalCount']
+
+def get_data(query, org):
+
+    # Hit the graphql endpoint and retry 3 times in case of failure
+    pc = PlatformConnector()
+    pc.logger = logging.getLogger()
+    pc.initialize_database_connections()
+
+    url = 'https://api.github.com/graphql'
```

Comment:
Double check if you need this or if its in the platform connector.

Reply:
It's how all the workers approach that. Can be improved for sure and put into a config, as the number of workers using it grows. 
If it's going to turn into a too-much-refactoring for this PR, this can be done separately.